src/logging.h

```diff
@@ -68,6 +69,106 @@ namespace BCLog {
         BLOCKSTORE  = (1 << 26),
         ALL         = ~(uint32_t)0,
     };
+
+
+    constexpr auto escape_sequence{"\033["};
+
+    class Color
+    {
+    public:
+        enum class Attribute {
+            /* Base attributes */
+            Reset = 0,
+            Bold = 1,
+            Faint = 2,
+            Italic = 3,
+            Underline = 4,
+            BlinkSlow = 5,
+            BlinkRapid = 6,
+            ReverseVideo = 7,
+            Concealed = 8,
+            CrossedOut = 9,
+
+            /* Foreground colors */
+            FgBlack = 30,
+            FgRed = 31,
+            FgGreen = 32,
+            FgYellow = 33,
+            FgBlue = 34,
+            FgMagenta = 35,
+            FgCyan = 36,
+            FgWhite = 37,
+
+            /* Foreground bright colors */
+            FgHiBlack = 90,
+            FgHiRed = 91,
+            FgHiGreen = 92,
+            FgHiYellow = 93,
+            FgHiBlue = 94,
+            FgHiMagenta = 95,
+            FgHiCyan = 96,
+            FgHiWhite = 97,
+
+            /* Background colors */
+            BgBlack = 40,
+            BgRed = 41,
+            BgGreen = 42,
+            BgYellow = 43,
+            BgBlue = 44,
+            BgMagenta = 45,
+            BgCyan = 46,
+            BgWhite = 47,
+
+            /* Background bright colors */
+            BgHiBlack = 100,
+            BgHiRed = 101,
+            BgHiGreen = 102,
+            BgHiYellow = 103,
+            BgHiBlue = 104,
+            BgHiMagenta = 105,
+            BgHiCyan = 106,
+            BgHiWhite = 107,
+        };
+
+        std::set<Attribute> attributes{};
+
+        Color() = delete;
+        Color(std::initializer_list<Attribute> attributes, const bool& enabled) : enabled(enabled)
+        {
+            for (auto attribute : attributes) {
+                this->attributes.insert(attribute);
+            }
+
+            m_sequence = this->ComputeSequence();
+        }
+
+        std::string Wrap(const std::string& str) const;
+        std::string operator()(const std::string& str) const { return Wrap(str); }
+
+        std::string m_sequence;
+        const bool& enabled;
+
+    private:
+        std::string ComputeSequence() const;
+    };
+
+    class ColorMan
+    {
+    public:
+        ColorMan()
+        {
+            m_enable_colors = getenv("TERM") != nullptr && getenv("TERM") != std::string("dumb") && (getenv("ANSI_COLORS_DISABLED") == nullptr || getenv("ANSI_COLORS_DISABLED") == std::string("0"));
```

Comment:
It might be ok to have colours on terminal by default, but not ok to have ANSI escape sequences written to debug.log by default. So at least `isatty()` check for stdout should be added. Or changed that colouring is always disabled by default and can be enabled manually.

Reply:
I just pushed a function that strips colors before writing to the debug file, is this right?